homeassistant/components/binary_sensor/ihc.py

```diff
@@ -0,0 +1,183 @@
+     'node': 'dataline_input',
+     'type': 'motion',
+     'inverting': False},
+    # Smoke detector
+    {'xpath': './/product_dataline[@product_identifier="_0x210a"]',
+     'node': 'dataline_input',
+     'type': 'smoke',
+     'inverting': False},
+    # leak detector
+    {'xpath': './/product_dataline[@product_identifier="_0x210c"]',
+     'node': 'dataline_input',
+     'type': 'moisture',
+     'inverting': False},
+    # light detector
+    {'xpath': './/product_dataline[@product_identifier="_0x2110"]',
+     'node': 'dataline_input',
+     'type': 'light',
+     'inverting': False},
+]
+
+_LOGGER = logging.getLogger(__name__)
+_IHCBINARYSENSORS = {}
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the IHC binary setsor platform."""
+    ihcplatform = get_ihc_platform(hass)
+    devices = []
+    if config.get(CONF_AUTOSETUP):
+        auto_setup(ihcplatform, devices)
+
+    binarysensors = config.get(CONF_BINARY_SENSORS)
+    if binarysensors:
+        _LOGGER.info("Adding IHC Binary Sensors")
+        for binarysensor in binarysensors:
+            ihcid = binarysensor[CONF_ID]
+            name = (binarysensor[CONF_NAME]
+                    if CONF_NAME in binarysensor
+                    else "ihc_" + str(ihcid))
+            sensortype = (binarysensor[CONF_TYPE]
+                          if CONF_TYPE in binarysensor
+                          else None)
+            inverting = (binarysensor[CONF_INVERTING]
+                         if CONF_INVERTING in binarysensor
+                         else False)
+            add_sensor(devices, ihcplatform.ihc, int(ihcid), name, sensortype,
+                       True, inverting)
+
+    add_devices(devices)
+    # Start notification after devices has been added
+    for device in devices:
+        device.ihc.add_notify_event(device.get_ihcid(),
+                                    device.on_ihc_change, True)
+
+
+def auto_setup(ihcplatform, devices):
+    """Auto setup ihc binary sensors from ihc project."""
+    _LOGGER.info("Auto setup - IHC Binary sensors")
+
+    def setup_product(ihcid, name, product, productcfg):
+        """Product setup callback."""
+        add_sensor_from_node(devices, ihcplatform.ihc, ihcid, name,
+                             product, productcfg['type'],
+                             productcfg['inverting'])
+    ihcplatform.autosetup(PRODUCTAUTOSETUP, setup_product)
+
+
+class IHCBinarySensor(IHCDevice, BinarySensorDevice):
+    """IHC Binary Sensor."""
+
+    def __init__(self, ihccontroller, name, ihcid, sensortype: str,
+                 inverting: bool, ihcname: str, ihcnote: str,
+                 ihcposition: str):
+        """Initialize the IHC binary sensor."""
+        IHCDevice.__init__(self, ihccontroller, name, ihcid, ihcname,
+                           ihcnote, ihcposition)
+        self._state = STATE_UNKNOWN
```

Comment:
I used STATE_UNKNOWN because I looked at binary_sensor/maxcube. I can see other platforms also using None, so I have changed it (I guess you know better than me)